src/Sfa.Tl.ResultsAndCertification.Api.Client/Clients/DfeSignInApiClient.cs

```diff
@@ -39,27 +42,32 @@ public async Task<DfeUserInfo> GetDfeSignInUserInfo(string organisationId, strin
 
             var userInfoResult = userInfo.Result;
             var ukprn = organisationUkprn.Result;
+            userInfoResult.Organisation = organisationUkprn.Result.Name;
 
-            if(ukprn.HasValue)
-                userInfoResult.Ukprn = ukprn;
+            if (ukprn.UKPRN.HasValue)
+                userInfoResult.Ukprn = ukprn.UKPRN;
+            else if (!ukprn.UKPRN.HasValue) userInfoResult.HasAccessToService = HasAccesstoService(Common.Constants.OrganisationConstants.AdminOrganisation, userInfoResult, RolesExtensions.AdminDashboardAccess);
             else
                 userInfoResult.HasAccessToService = false;
             
             return userInfoResult;
         }
 
-        private async Task<long?> GetOrganisationUkprn(string organisationId, string userId)
+        private async Task<Organisation> GetOrganisationUkprn(string organisationId, string userId)
         {
+            var organisation = new Organisation();
             var requestUri = $"/users/{userId}/organisations";
             var response = await _httpClient.GetAsync(requestUri);
 
             if (response.IsSuccessStatusCode)
             {
                 var responseContent = await response.Content.ReadAsStringAsync();
                 var orgToken = JArray.Parse(responseContent).FirstOrDefault(org => org.SelectToken("id").ToString() == organisationId);
-                return orgToken?["ukprn"].ToObject<long?>();
+                organisation.UKPRN = (int?)(orgToken?["ukprn"]);
+                organisation.Name = (string)(orgToken?["name"]);
+                return organisation;
```

Comment:
This line can be removed. The object will be returned in the last line

Reply:
Line 60 is removed already and that is change